tests/subsys/net/lib/aws_fota/aws_fota_json/src/main.c

```diff
@@ -33,20 +33,85 @@ static void test_parse_job_execution(void)
 						      job_id, hostname,
 						      file_path,
 						      &version_number);
-
 	zassert_equal(ret, 1, NULL);
+
 	zassert_true(!strcmp(job_id, expected_job_id), NULL);
 	zassert_equal(version_number, expected_version_number, NULL);
 	zassert_true(!strcmp(hostname, expected_hostname), NULL);
 	zassert_true(!strcmp(file_path, expected_file_path), NULL);
 }
 
-static void test_parse_job_execution_missing_field(void)
+static void test_parse_malformed_job_execution(void)
 {
-	char expected_job_id[] = "9b5caac6-3e8a-45dd-9273-c1b995762f4a";
-	char expected_hostname[] = "fota-update-bucket.s3.eu-central-1.amazonaws.com";
-	char expected_file_path[] = "/update.bin?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAWXEL53DXIU7W72AE%2F20190606%2Feu-central-1%2Fs3%2Faws4_request&X-Amz-Date=20190606T081505Z&X-Amz-Expires=604800&X-Amz-Signature=913e00b97efe5565a901df4ff0b87e4878a406941d711f59d45915035989adcc&X-Amz-SignedHeaders=host";
+	int ret;
+	char job_id[100];
+	char hostname[100];
+	char file_path[1000];
+	int version_number;
 
+	char malformed[] = "{\"timestamp\":1559808907,\"execution\":{jobId\":\"9b5caac6-3e8a-45dd-9273-c1b995762f4a\",\"status\":\"QUEUED\"\"queuedAt\":1559808906,\"lastUpdatedAt\":1559808906,\"versionNumber\":1,\"executionNumber\":1,\"jobDocument\":{\"operation\":\"app_fw_update\",\"fwversion\":\"2\",\"size\":181124,\"location\":{\"protocol\":\"https:\",\"host\":\"fota-update-bucket.s3.eu-central-1.amazonaws.com\",\"path\":\"/update.bin?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAWXEL53DXIU7W72AE%2F20190606%2Feu-central-1%2Fs3%2Faws4_request&X-Amz-Date=20190606T081505Z&X-Amz-Expires=604800&X-Amz-Signature=913e00b97efe5565a901df4ff0b87e4878a406941d711f59d45915035989adcc&X-Amz-SignedHeaders=host\"}}}}";
+/* Memset to ensure correct null-termination */
```

Comment:
You might want to align this comment. Or, on the other hand, are these memsets really needed here? You're not verifying the content of the strings here, hence they only make the test code unnecessarily longer (and harder to understand).

Reply:
Same comment applies for other new tests as well of course.

Reply:
I don't think the memsets are needed anymore maybe needed for zassert but I'll try to remove them.